evalml/automl/automl_search.py

```diff
@@ -1276,7 +1276,7 @@ def get_pipeline(self, pipeline_id):
             )
         new_pipeline = pipeline.new(parameters, random_seed=self.random_seed)
         if is_binary(self.problem_type):
-            new_pipeline.threshold = pipeline.threshold
+            new_pipeline.threshold = None
```

Comment:
I think it might be better to take the average threshold over none? My concern with setting it to None is that it won’t reflect the threshold used to compute the scores in the leaderboard. In my opinion, taking the average threshold is closer to matching the mean_cv_score we use to sort the leaderboard.